hudi-spark-datasource/hudi-spark-common/src/main/scala/org/apache/hudi/HoodieSparkSqlWriter.scala

```diff
@@ -1018,6 +1018,11 @@ class HoodieSparkSqlWriterInternal {
       val metaSyncSuccess = metaSync(spark, HoodieWriterUtils.convertMapToHoodieConfig(parameters),
         tableInstantInfo.basePath, schema)
 
+      if (!metaSyncSuccess && commitSuccess) {
+        log.error("Metadata sync operation failed, rolling back current commit")
+        client.rollback(tableInstantInfo.instantTime)
```

Comment:
I don't think rollback should be done if meta sync has failure.  Hudi is designed in a way that transaction and meta sync are decoupled.  User can read a table through the base path with or without meta / catalog sync

Reply:
I get the idea. In some context described in the issue unless you rollback the table gets not readible from engine. Rollback could be configurable otwz ?

Reply:
See my comment in the GH issue.  If required, we can make this configurable; but the default behavior should not roll back the completed transaction.